cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1075,6 +1094,137 @@ protected StatefulSet createStatefulSet(
         return statefulSet;
     }
 
+    /**
+     * Creates the StrimziPodSet with the Pods which currently correspond to the existing StatefulSet pods.
+     *
+     * @param replicas          Defines how many pods should be generated and stored in this StrimziPodSet
+     * @param setAnnotations    Map with annotations which should be set on the StrimziPodSet
+     * @param podAnnotations    Map with annotation which should be set on the Pods
+     * @param volumes           Function which returns a list of volumes which should be used by the Pod and its containers
+     * @param affinity          Affinity rules for the pods
+     * @param initContainers    List of init containers which should be used in the pods
+     * @param containers        List of containers which should be used in the pods
+     * @param imagePullSecrets  List of image pull secrets with container registry credentials
+     * @param isOpenShift       Flag to specify whether we are on OpenShift or not
+     *
+     * @return                  Generated StrimziPodSet with all pods
+     */
+    protected StrimziPodSet createPodSet(
+            int replicas,
+            Map<String, String> setAnnotations,
+            Map<String, String> podAnnotations,
+            Function<String, List<Volume>> volumes,
+            Affinity affinity,
+            List<Container> initContainers,
+            List<Container> containers,
+            List<LocalObjectReference> imagePullSecrets,
+            boolean isOpenShift)  {
+        List<Map<String, Object>> pods = new ArrayList<>(replicas);
+
+        for (int i = 0; i < replicas; i++)  {
+            String podName = ZookeeperCluster.zookeeperPodName(cluster, i);
```

Comment:
I know that this PR is just for removing SS for ZooKeeper and not for Kafka but could we try to make this method more general for using for the future SS removal for Kafka as well? It's even in the `AbstractModel` class that should not have any specific references to ZK or Kafka. Maybe we could pass a Java `Function` to provide the pod name?

Reply:
Good point. Fixed.